lib/ContextModule.js

```diff
@@ -144,7 +144,7 @@ class ContextModule extends Module {
 	prettyRegExp(regexString) {
 		// remove the "/" at the front and the beginning
 		// "/foo/" -> "foo"
-		return regexString.substring(1, regexString.length - 1);
+		return regexString.substring(1, regexString.length - 1).replace("!", "%21");
```

Comment:
```suggestion
		return regexString.substring(1, regexString.length - 1).replace(/!/g, "%21");
```

Reply:
> Please add a test case that covers this case.
> 
> Take a look on `require.context` test cases https://github.com/webpack/webpack/tree/master/test/cases/context

@sokra / @vankop `require.context()` just export three members `keys()`, `id` and `resolve()`, how can I get the `sourcemap` in tests?.